### PR TITLE
extend/refactor S42IslRttSpec

### PR DESCRIPTION
**Pull request by andriidovhan**

### CODE REVIEW

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42IslRttSpec.groovy [hunk 1]
```diff
@@ -1,358 +1,714 @@
...
+                assert !islUtils.getIslInfo(newIsl.reversed).isPresent()
+            }
+        }
+        database.resetCosts()
+    }
+
+    @Tidy
+    @Ignore
+    @Tags([HARDWARE])//it works on hardware env only
+    def "ISL RTT stats are available for one direction in case link is UP in forward direction only"() {
```
Comment (sergii-iakovenko) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42IslRttSpec.groovy:276]: I don't think we will support the case: ISL RTT relies on both directions and wont work if opposite one is down.

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42IslRttSpec.groovy [hunk 2]
```diff
@@ -383,9 +745,44 @@ class Server42IslRttSpec extends HealthCheckSpecification {
...
+        def stats = otsdb.query(t, new Date(), metricPrefix + "isl.rtt",
+                [src_switch: isl.srcSwitch.dpId.toOtsdFormat(),
+                 src_port  : String.valueOf(isl.srcPort),
+                 dst_switch: isl.dstSwitch.dpId.toOtsdFormat(),
+                 dst_port  : String.valueOf(isl.dstPort),
+                 origin    : "server42"]).dps
+        def expected = stats*.getValue().sum()/stats.size()
+        def actual = northbound.getLink(isl).latency
+        log.info("latencyValue expected: $expected")
+        log.info("latencyValue actual: $actual")
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42IslRttSpec.groovy:784]: remove?

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42IslRttSpec.groovy [hunk 3]
```diff
@@ -383,9 +745,44 @@ class Server42IslRttSpec extends HealthCheckSpecification {
...
+    void verifyLatencyValueIsCorrect(Isl isl, Boolean valueIsCorrect){
+        def t = new Date()
+        t.setSeconds(t.getSeconds() - 600) //kilda_latency_update_time_range: 600
+        def stats = otsdb.query(t, new Date(), metricPrefix + "isl.rtt",
+                [src_switch: isl.srcSwitch.dpId.toOtsdFormat(),
+                 src_port  : String.valueOf(isl.srcPort),
+                 dst_switch: isl.dstSwitch.dpId.toOtsdFormat(),
+                 dst_port  : String.valueOf(isl.dstPort),
+                 origin    : "server42"]).dps
+        def expected = stats*.getValue().sum()/stats.size()
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42IslRttSpec.groovy:781]: you can do `stats.average { it.value }`

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42IslRttSpec.groovy [hunk 4]
```diff
@@ -383,9 +745,44 @@ class Server42IslRttSpec extends HealthCheckSpecification {
...
+        def stats = otsdb.query(checkpointTime, metricPrefix + "isl.rtt",
+                [src_switch: isl.srcSwitch.dpId.toOtsdFormat(),
+                 src_port  : String.valueOf(isl.srcPort),
+                 dst_switch: isl.dstSwitch.dpId.toOtsdFormat(),
+                 dst_port  : String.valueOf(isl.dstPort),
+                 origin    : "server42"]).dps
+        assert statExist ? !stats.isEmpty() : stats.isEmpty()
+    }
+
+    void verifyLatencyValueIsCorrect(Isl isl, Boolean valueIsCorrect){
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42IslRttSpec.groovy:772]: not sure how `Boolean valueIsCorrect` should be used

**Pull request closed without merging**